music_assistant/controllers/streams.py

```diff
@@ -919,25 +919,39 @@ async def get_announcement_stream(
         use_pre_announce: bool = False,
     ) -> AsyncGenerator[bytes, None]:
         """Get the special announcement stream."""
+        filter_params = ["loudnorm=I=-10:LRA=11:TP=-2"]
+
+        if use_pre_announce:
+            # Note: TTS URLs might take a while to load cause the actual data are often generated
+            # asynchronously by the TTS provider. If we ask ffmpeg to mix the pre-announce, it will
+            # wait until it reads the TTS data, so the whole stream will be delayed. It is much
+            # faster to first play the pre-announce using a separate ffmpeg stream, and only
+            # afterwards play the TTS itself.
+            #
+            # For this to be effective the player itself needs to be able to start playback fast.
+            # If the returned stream is used as input to ffmpeg we should pass -probesize 8096.
+            #
+            # Finally we also need to make sure we don't make other blocking requests to the TTS
+            # data, eg to get the duration (async_parse_tags).
+            #
+            async for chunk in get_ffmpeg_stream(
+                audio_input=ANNOUNCE_ALERT_FILE,
+                input_format=AudioFormat(content_type=ContentType.try_parse(ANNOUNCE_ALERT_FILE)),
+                output_format=output_format,
+                filter_params=filter_params,
+            ):
+                yield chunk
+
```

Comment:
This will not work in all cases because a player might detect EOF and stop playing. Did you try this with a couple of different players ? Especially players that get the url themselves ?

Reply:
Not really, let me test before merging and/or think alternative solutions.

I tried to do it within ffmpeg, that is convince it to start merging A and B before B arrives, but it might be impossible before reading even a single byte of B.

Reply:
I did some base tests with a few players and it seems to work just fine. So let's keep it as-is and change only if we get issue reports